netmiko/base_connection.py

```diff
@@ -105,15 +107,19 @@ def __init__(self, ip='', host='', username='', password='', secret='', port=Non
             self.ip = ip
         elif host:
             self.host = host
-        if not ip and not host:
+        if not ip and not host and 'serial' not in device_type:
             raise ValueError("Either ip or host must be set")
         if port is None:
             if 'telnet' in device_type:
                 self.port = 23
             else:
                 self.port = 22
         else:
-            self.port = int(port)
+            if 'serial' in device_type:
+                self.port = check_serial_port(port)
```

Comment:
It looks like here you are overloading the meaning of port? In other words, we are normally using this for TCP port 22 and TCP port 23, but here you are switching it to mean 'com1', 'com2', et cetera?

Is that right?

Reply:
Yes.

Reply:
Okay, I separated that out such that you would pass a 'port' key inside the `serial_settings`.